auth-request.lua

```diff
@@ -89,33 +139,57 @@ core.register_action("auth-request", { "http-req" }, function(txn, be, path)
 	end
 
 	-- Make request to backend.
-	local response, err = http.head {
+	if method == "*" then
+		method = txn.sf:method()
+	end
+	local response, err = http.send(method:upper(), {
 		url = "http://" .. addr .. path,
 		headers = headers,
-	}
+	})
+
+	-- `terminate_on_failure == true` means that the Lua script should send the response
+	-- and terminate the transaction in the case of a failure. This will happen when
+	-- hdr_fail content isn't a dash `-`.
+	local terminate_on_failure = hdr_fail ~= "-"
 
 	-- Check whether we received a valid HTTP response.
 	if response == nil then
 		txn:Warning("Failure in auth-request backend '" .. be .. "': " .. err)
-		set_var(txn, "txn.auth_response_code", 500)
+		if terminate_on_failure then
+			send_response(txn)
+		else
+			set_var(txn, "txn.auth_response_code", 500)
+		end
```

Comment:
```suggestion
		set_var(txn, "txn.auth_response_code", 500)
		if terminate_on_failure then
			send_response(txn)
		end
```
I would put setting the variable unconditionally before the `if` where it was before. It doesn't really hurt and keeps the control flow a bit simpler.